File: min-platform-ui/src/utils/enums/oldNoticeEnums.js
import {byValue} from "./enums";




export const noticeTypeEnum = {
  NOTICE: {
    desc: "通知",
    value: "NOTICE"
  },
  ANNOUNCEMENT: {
    desc: "公告",
    value: "ANNOUNCEMENT"
  },
  values() {
    return [
      this.NOTICE,
      this.ANNOUNCEMENT,
    ]
  },
  byValue(value) {
    return byValue(this.values(), value);
  },
};



export const noticeStatusEnum = {
  NORMAL: {
    desc: "正常",
    value: "NORMAL"
  },
  CLOSE: {
    desc: "关闭",
    value: "CLOSE"
  },
  values() {
    return [
      this.NORMAL,
      this.CLOSE,
    ]
  },
  byValue(value) {
    return byValue(this.values(), value);
  },
};

File: old-modules/old-module-business/src/main/java/com/old/business/user/enums/user/OldNoticeEnums.java
package com.old.business.user.enums.user;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

/**
 * 通知公告表 实体的枚举类 old_notice
 *
 * @author old
 * @date 2023-05-20
 */
public interface OldNoticeEnums {


    @Getter
    @ToString
    @RequiredArgsConstructor
    enum NoticeType {
        // 公告类型（通知 公告）
        NOTICE("通知"),
        ANNOUNCEMENT("公告"),
        ;

        private final String desc;

    }

    @Getter
    @ToString
    @RequiredArgsConstructor
    enum NoticeStatus {
        // 公告状态（0正常 1关闭）
        NORMAL("正常"),
        CLOSE("关闭"),
        ;

        private final String desc;

    }

}
